src/idl_gen_grpc.cpp

```diff
@@ -173,45 +176,168 @@ class FlatBufFile : public grpc_generator::File {
+  }
+};
+
+// Gets the Codec for the flatbuffers implementation of Codec interface grpc-go
+inline std::string GetGoCodec() {
+  std::string codec;
+  auto printer = std::unique_ptr<grpc_cpp::FlatBufPrinter>(new grpc_cpp::FlatBufPrinter(&codec));
+  printer->Print("// FlatCodec implements gRPC-go Codec\n\n");
+  printer->Print("var Codec string = \"flatbuffers\"\n\n");
+  printer->Print("type FlatCodec struct{}\n\n");
+  printer->Print("func (FlatCodec) Marshal(v interface{}) ([]byte, error) {\n");
+  printer->Indent();
+  printer->Print("return v.(*flatbuffers.Builder).FinishedBytes(), nil\n");
+  printer->Outdent();
+  printer->Print("}\n\n");
+  printer->Print("func (FlatCodec) Unmarshal(data []byte, v interface{}) error {\n");
+  printer->Indent();
+  printer->Print("v.(flatbuffersInit).Init(data, flatbuffers.GetUOffsetT(data))\n");
+  printer->Print("return nil\n");
+  printer->Outdent();
+  printer->Print("}\n\n");
+  printer->Print("func (FlatCodec) String() string {\n");
+  printer->Indent();
+  printer->Print("return Codec\n");
+  printer->Outdent();
+  printer->Print("}\n\n");
+  printer->Print("type flatbuffersInit interface {\n");
+  printer->Indent();
+  printer->Print("Init(data []byte, i flatbuffers.UOffsetT)\n");
+  printer->Outdent();
+  printer->Print("}\n\n");
+  return codec;
+}
+
+class GoGrpcGenerator : public flatbuffers::BaseGenerator {
+ public:
+  GoGrpcGenerator(const Parser &parser, const std::string &path,
+                  const std::string &file_name)
+    : BaseGenerator(parser, path, file_name, "", "" /*Unused*/),
+      parser_(parser), path_(path), file_name_(file_name) {}
+
+  //Generates services into different files
+  bool generate() {
+    FlatBufFile file(parser_, file_name_);
+    grpc_go_generator::Parameters p;
+    p.custom_method_input = "flatbuffers.Builder";
+    //To make sure the flatbuffers codec is generated only once in a package
+    std::map<std::string, bool> uniquePackages;
+    for (int i = 0; i < file.service_count(); i++) {
+      auto service = file.service(i);
+      const Definition *def = parser_.services_.vec[i];
+      std:: string package_name = LastNamespacePart(*(def->defined_namespace));
+      if (!uniquePackages[package_name]) {
+        p.custom_codec_source = GetGoCodec();
+        uniquePackages[package_name] = true;
+      } else {
+        p.custom_codec_source = "";
+      }
+      p.package_name = package_name;
+      std::string output = grpc_go_generator::GenerateServiceSource(&file, service.get(), &p);
+      std::string filename = NamespaceDir(*def->defined_namespace) + def->name + "_grpc.go";
+      if (!flatbuffers::SaveFile(filename.c_str(), output, false))
+        return false;
+    }
+    return true;
+  }
+
+ protected:
+  const Parser &parser_;
+  const std::string &path_, &file_name_;
+};
+}//namespace grpc_go
+
+bool GenerateGRPC(const Parser &parser,
+                  const std::string &path,
```

Comment:
My bad, will try to avoid it in future. I had done the same thing as seen in the python PR but was sceptical about, anyway I'll quickly fix this and get back to you.